interpreter/exec/simd.ml

```diff
@@ -72,26 +79,49 @@ struct
   let i32x4_extract_lane i x = List.nth (to_i32x4 x) i
 
   let to_f32x4 = Rep.to_f32x4
-  let of_f32x4 = Rep.of_f32x4
-  let f32x4_unop f x =
-    of_f32x4 (List.map f (to_f32x4 x))
-  let f32x4_binop f x y =
-    of_f32x4 (List.map2 f (to_f32x4 x) (to_f32x4 y))
 
   let f32x4_extract_lane i x = List.nth (to_f32x4 x) i
-  let f32x4_min = f32x4_binop F32.min
-  let f32x4_max = f32x4_binop F32.max
-  let f32x4_abs x = f32x4_unop F32.abs x
 
   let to_f64x2 = Rep.to_f64x2
-  let of_f64x2 = Rep.of_f64x2
-  let f64x2_unop f x =
-    of_f64x2 (List.map f (to_f64x2 x))
-  let f64x2_binop f x y =
-    of_f64x2 (List.map2 f (to_f64x2 x) (to_f64x2 y))
 
   let f64x2_extract_lane i x = List.nth (to_f64x2 x) i
-  let f64x2_min = f64x2_binop F64.min
-  let f64x2_max = f64x2_binop F64.max
-  let f64x2_abs x = f64x2_unop F64.abs x
+
+  module type Shape =
```

Comment:
@rossberg does this look like something you had in mind?
I had to duplicate this `Shape` sig, not sure how else to do it.
And the other problem I had with this nested module is that. `Shape.u` is now opaque, so I couldn't really implement extract_lane
```
module F32x4 : Shape = struct
type u = F32.t
val extract_lane (i : int) (x : t) : u = implementation
```
Then callers will get a `Simd.F32x4.u` instead of a `F32.t`.

Reply:
Found a way around this in the latest commit.